pennylane/fourier/utils.py

```diff
@@ -63,8 +63,7 @@ def get_spectrum(op, decimals):
     Returns:
         set[float]: non-negative frequencies contributed by this input-encoding gate
     """
-    matrix, coeff = get_generator(op, return_matrix=True)
-    matrix = coeff * matrix
+    matrix = qml.matrix(qml.generator(op, format="observable"))
```

Comment:
This could have been done in #2182 already, but now that we're touching this code, we could simply use `parameter_frequencies` instead of this whole utility function? Might be a separate PR, though (maybe together with Rotosolve using `parameter_frequencies`, grouping it into an "application" PR).

Reply:
good idea!

Reply:
@dwierichs which would you prefer, doing it here, or doing it in a separate PR?

Reply:
I guess a separate one that uses `parameter_frequencies` where applicable is more clean? :)

Reply:
yep :)